app/src/main/java/edu/uw/myuw_android/UserInfoStore.kt

```diff
@@ -0,0 +1,164 @@
+            ),
+            "Accounts" to Triple(
+                R.id.nav_accounts,
+                R.drawable.ic_accounts,
+                { _ -> true }
+            ),
+            "Notices" to Triple(
+                R.id.nav_notices,
+                R.drawable.ic_notices,
+                { aff: List<String> -> aff.contains("student") }
+            ),
+            "Profile" to Triple(
+                R.id.nav_profile,
+                R.drawable.ic_account,
+                { _ -> true }
+            ),
+            "Calendar" to Triple(
+                R.id.nav_academic_calendar,
+                R.drawable.ic_calander,
+                { aff: List<String> -> true }
+            ),
+            "UW Resources" to Triple(
+                R.id.nav_resources,
+                R.drawable.ic_resources,
+                { _ -> true }
+            )
+        ),
+        R.id.group_nav_drawer_secondary to mapOf(
+            "Sign out" to Triple(
+                R.id.logout,
+                R.drawable.ic_exit_to_app_black_24dp,
+                { _ -> true }
+            )
+        )
+    )
+
+    private var activeMenuItems = menuItems
+    private var affiliations = mutableListOf<String>()
+
+    var name: MutableLiveData<String> = MutableLiveData()
+    var email: MutableLiveData<String>  = MutableLiveData()
+    var netId: MutableLiveData<String> = MutableLiveData()
+    var emailForwardUrl: MutableLiveData<String> = MutableLiveData()
+    var groups: MutableLiveData<Set<String>> = MutableLiveData()
+
+    fun setNavigationMenu(menu: Menu, context: Context) {
+        menu.clear()
+
+        Log.d("setNavigationMenu - affiliations: ", affiliations.toString())
+        for (key in activeMenuItems.keys) {
+            val currentGroup = activeMenuItems.getValue(key)
+
+            menu.setGroupCheckable(key, true, true)
+            for (title in currentGroup.keys) {
+                Log.d("setNavigationMenu - $title", currentGroup.getValue(title).third(affiliations).toString())
+                if (currentGroup.getValue(title).third(affiliations)) {
+                    val menuItem = menu.add(
+                        key,
+                        currentGroup.getValue(title).first,
+                        Menu.NONE,
+                        title
+                    )
+                    menuItem.icon = ResourcesCompat.getDrawable(
+                        context.resources,
+                        currentGroup.getValue(title).second,
+                        null
+                    )
+                }
+            }
+        }
+    }
+
+    fun updateAffiliations(resources: Resources, idToken: String) {
+        val conn =
+            URL(resources.getString(R.string.myuw_affiliation_endpoint)).openConnection()
+        conn.setRequestProperty("Authorization", "Bearer $idToken")
+        if (conn !is HttpsURLConnection && !BuildConfig.DEBUG) throw SecurityException("Connection is not secure")
+
```

Comment:
I see you are using HttpsURLConnection to make network calls here. But in the industry, Retrofit is a much more popular choice since it would make the code a lot more readable and faster the process. And here's a link for the reason why they use it http://instructure.github.io/blog/2013/12/09/volley-vs-retrofit/